packages/shared/src/components/comments/CommentInputOrPage.tsx

```diff
@@ -0,0 +1,48 @@
+import React, { ReactElement } from 'react';
+import { useRouter } from 'next/router';
+import {
+  CommentMarkdownInput,
+  CommentMarkdownInputProps,
+} from '../fields/MarkdownInput/CommentMarkdownInput';
+import { ViewSize, useViewSize } from '../../hooks';
+
+interface CommentInputOrPageProps
+  extends Omit<CommentMarkdownInputProps, 'className'> {
+  onClose?: () => void;
+  className?: {
+    input?: CommentMarkdownInputProps['className'];
+    modal?: string;
+  };
+  replyToCommentId?: string;
+}
+
+export default function CommentInputOrPage({
```

Comment:
Component responsible for either rendering a comment input or redirect to a nextjs page instead. The redirect only happens on mobile.